crates/autopilot/src/decoded_settlement.rs

```diff
@@ -138,15 +137,13 @@ impl From<(Address, U256, Bytes<Vec<u8>>)> for DecodedInteraction {
 #[derive(Debug, Clone)]
 pub struct OrderExecution {
     pub order_uid: OrderUid,
-    pub executed_solver_fee: Option<U256>,
+    pub executed_fee: Option<U256>,
```

Comment:
Why does this need to be `Option`?

Reply:
Looks like `on_settlement_event_updater` (which updates the required DB entries when a settlement gets indexed) requires 2 pieces of information:
1. the full (signed) fee amount collected in total by the settlement
2. all `OrderExecution`s that don't have a fee assigned to them yet

These pieces of information can currently be served by the same function `Postgres::order_executions_for_tx()` which returns all `OrderExecution`s (with and without a signed fee). For `1` we currently sum all the fees which are already populated and for 2 we use only the executions where `executed_fee.is_none()`.

So refactoring this field to be non-optional might make other code weirder.

While looking into that I stumbled over this [code](https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/on_settlement_event_updater.rs#L204). Could somebody give that a sanity check. My current intuition is that that code is correct because it should only contain the fees that were signed upfront by the user because those are the only fees that are part of the objective value computation. But because fees have become so complicated since we introduced fees that were not signed upfront I think checking that logic once more might be good.